src/colorizer/utils/data_utils.ts

```diff
@@ -327,3 +328,41 @@ export function getGlobalIdFromSegId(
 
   return rawGlobalId - 1; // -1 to convert to zero-based index
 }
+
+export function getLabelTypeFromParsedCsv(
+  headers: string[],
+  data: Record<string, string | undefined>[]
+): Map<string, LabelType> {
+  const labelTypeMap = new Map<string, LabelType>();
+  for (const header of headers) {
+    let hasIntegerValues = true;
+    let hasBooleanValues = true;
+    for (const row of data) {
+      const value = row[header]?.trim();
+      const valueAsInt = parseInt(value ?? "", 10);
+      if (value === undefined || value === "") {
+        continue;
+      } else if (value.toLowerCase() === BOOLEAN_VALUE_TRUE || value.toLowerCase() === BOOLEAN_VALUE_FALSE) {
+        hasIntegerValues = false;
+      } else if (valueAsInt.toString(10) === value && Number.isInteger(valueAsInt)) {
```

Comment:
It's not most intuitive why we parsed to integer in L342 and then converting back to string here. Adding a clarifying comment?